toolchain/lexer/tokenized_buffer.cpp

```diff
@@ -111,39 +111,39 @@ static auto ScanForIdentifierPrefix(llvm::StringRef text) -> llvm::StringRef {
   //
   // bits 4-7 remain unused if we need to classify more characters.
   const auto high_lut = _mm_setr_epi8(
-      /*0x0:*/ 0b0000'0000,
-      /*0x1:*/ 0b0000'0000,
-      /*0x2:*/ 0b0000'0000,
-      /*0x3:*/ 0b0000'0010,
-      /*0x4:*/ 0b0000'0100,
-      /*0x5:*/ 0b0000'1001,
-      /*0x6:*/ 0b0000'0100,
-      /*0x7:*/ 0b0000'1000,
-      /*0x8:*/ 0b0000'0000,
-      /*0x9:*/ 0b0000'0000,
-      /*0xA:*/ 0b0000'0000,
-      /*0xB:*/ 0b0000'0000,
-      /*0xC:*/ 0b0000'0000,
-      /*0xD:*/ 0b0000'0000,
-      /*0xE:*/ 0b0000'0000,
-      /*0xF:*/ 0b0000'0000);
+      /* __b0=*/0b0000'0000,
```

Comment:
I'm a bit hesitant to add comments listing the names these parameters happen to have in one "random" compiler's `<emmintrin.h>`.

These names don't match the ones from [intel's intrinsics guide](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_setr_epi8) where they're named `e15`, `e14`, ..., `e0`, nor the one in intel's Software Development Manual, volume 2C (page C-17, table C-2), where they're named `w15`, `w14`, ..., `w0`. And they don't match the names used by GCC's `<emmintrin.h>`, which are `__q00`, ..., `__q15`.

I suppose the point here is that these match the names used specifically in the header that `clang-tidy` is looking at, so maybe they're the best we can do if we want any kind of checking?

Reply:
Sadly, we don't get clang-tidy checking for these either. I tried a bunch of spacing and other things, none got a clang-tidy issue when I wrote an actual bug.

Reply:
The check intentionally ignores parameter names from system headers because they're definitely not part of the API contract and likely to be reserved identifiers: https://github.com/llvm/llvm-project/blob/2b664d678dcd6b23eda992b45d009788e34ea8f1/clang-tools-extra/clang-tidy/bugprone/ArgumentCommentCheck.cpp#L67